packages/sdk-snippets/src/index.ts

```diff
@@ -0,0 +1,53 @@
+import type { ClientId, SnippetType, TargetId } from './targets/targets';
+
+import { targets } from './targets/targets';
+
+export { availableServerTargets, availableWebhookTargets, extname } from './helpers/utils';
+
+export interface ServerParameter {
+  name: string;
+  default: string;
+  source: 'server';
+}
+
+export interface SecurityParameter {
+  name: string;
+  default: string;
+  source: 'security';
+  type: string;
+}
+
+export type Parameters = (ServerParameter | SecurityParameter)[];
+
+export class MetricsSDKSnippet {
+  params: Parameters = [];
+
+  constructor(input: Parameters) {
+    this.params = input;
+  }
+
+  convert = (snippetType: SnippetType, targetId: TargetId, clientId?: ClientId, options?: any) => {
+    if (!options && clientId) {
+      // eslint-disable-next-line no-param-reassign
+      options = clientId;
+    }
+
+    const target = targets[targetId];
+    if (!target || !target.services.webhooks) {
```

Comment:
Are there any chances these object children could be `undefined`? Or would that get caught in the PR process for future target/client additions?

Reply:
That'd get caught by TS as `services.webhooks` is required.